src/JWT/Builder/EnumExtensions.cs

```diff
@@ -20,10 +20,13 @@ internal static class EnumExtensions
         /// <summary>
         /// Gets the value of the <see cref="DescriptionAttribute" /> from the object.
         /// </summary>
-        private static string GetDescription<T>(T value) =>
-            value.GetType()
-                 .GetField(value.ToString())
-                 .GetCustomAttribute<DescriptionAttribute>()
-                ?.Description ?? value.ToString();
+        private static string GetDescription<T>(T value)
+        {

```

Comment:
Can we take the advantage of relatively-new constraint `where T : enum` here and do something more optimal?

Reply:
Short answer: there is no benefit of the proposed change

Long answer:
`where T : enum` does not compile. I checked two options:
- `where T : Enum`
- `where T : struct`

results of benchmarks:
```
BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19042
Intel Core i7-9750H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=5.0.201
[Host]     : .NET Core 5.0.4 (CoreCLR 5.0.421.11614, CoreFX 5.0.421.11614), X64 RyuJIT
DefaultJob : .NET Core 5.0.4 (CoreCLR 5.0.421.11614, CoreFX 5.0.421.11614), X64 RyuJIT

|             Method |     Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------- |---------:|---------:|---------:|-------:|------:|------:|----------:|
|            Current | 984.4 ns | 16.97 ns | 15.88 ns | 0.0381 |     - |     - |     248 B |
|   Enum_Constrained | 990.9 ns | 18.60 ns | 16.49 ns | 0.0381 |     - |     - |     248 B |
| Struct_Constrained | 983.2 ns | 16.67 ns | 15.59 ns | 0.0381 |     - |     - |     248 B |
```
I used the newest version of .NET Core to test because it has the most optimisation hacks in Roslyn/JIT.

As you can see, the results are pretty the same.  
Also IL code built in Release mode does not have important differences.

Here is benchmark tests that I used to compare the solutions:
```c#
  [MemoryDiagnoser]
  public class EnumDescriptionBenchmark
  {
    [Benchmark]
    public string Current() => GetDescription(HeaderName.X5c);

    [Benchmark]
    public string Enum_Constrained() => GetDescription_Enum_Constrained(HeaderName.X5c);

    [Benchmark]
    public string Struct_Constrained() => GetDescription_Struct_Constrained(HeaderName.X5c);

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static string GetDescription<T>(T value) =>
      value.GetType()
           .GetField(value.ToString())
           .GetCustomAttribute<DescriptionAttribute>()
          ?.Description ?? value.ToString();

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static string GetDescription_Enum_Constrained<T>(T value) where T : Enum =>
      value.GetType()
           .GetField(value.ToString())
           .GetCustomAttribute<DescriptionAttribute>()
          ?.Description ?? value.ToString();

    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static string GetDescription_Struct_Constrained<T>(T value) where T : struct =>
      value.GetType()
           .GetField(value.ToString())
           .GetCustomAttribute<DescriptionAttribute>()
          ?.Description ?? value.ToString();

    private enum HeaderName
    {
      [Description("typ")]
      Type,

      [Description("cty")]
      ContentType,

      [Description("alg")]
      Algorithm,

      [Description("kid")]
      KeyId,

      [Description("x5u")]
      X5u,

      [Description("x5c")]
      X5c,

      [Description("x5t")]
      X5t
    }
  }
```